Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/Annotation/RemoveIshAnnotation.cs

```diff
@@ -0,0 +1,161 @@
+            if (IshSession == null) { throw new ArgumentException(ISHRemoteSessionStateIshSessionException); }
+            WriteDebug($"Using IshSession[{IshSession.Name}] from SessionState.{ISHRemoteSessionStateIshSession}");
+
+            if ((IshSession.ServerIshVersion.MajorVersion < 14) || ((IshSession.ServerIshVersion.MajorVersion == 14) && (IshSession.ServerIshVersion.RevisionVersion < 1)))
+            {
+                throw new PlatformNotSupportedException($"Remove-IshAnnotation requires server-side Annotation API which is only available starting from 14.0.1 and up. ServerIshVersion[{IshSession.ServerVersion}]");
+            }
+            base.BeginProcessing();
+        }
+
+        /// <summary>
+        /// Process the Remove-IshAnnotation commandlet.
+        /// </summary>
+        /// <exception cref="TrisoftAutomationException"></exception>
+        /// <exception cref="Exception"></exception>
+        protected override void ProcessRecord()
+        {
+            try
+            {
+                WriteDebug("Deleting");
+                List<string> annotationIdsToDelete = new List<string>();
+
+                //1. Prepare list of annotations to delete depending on the chosen ParameterSet
+                switch(ParameterSetName)
+                {
+                    case "ParametersGroup":
+                        annotationIdsToDelete.Add(AnnotationId);
+                        break;
+                    case "IshAnnotationGroup":
+                        annotationIdsToDelete.AddRange(Array.ConvertAll(IshAnnotation, ishAnnotation => ishAnnotation.IshRef));
+                        break;
+                }
+
+                //2. Delete required annotations
+                //2.1 Check if Force is enabled (so need to delete replies)
+                if (Force)
+                {
+                    // find replies for the required annotation(s)
+                    IshFields requestedMetadata = new IshFields();
+                    List<IshObject> ishObjectsList = new List<IshObject>();
+
+                    requestedMetadata.AddField(new IshRequestedMetadataField(FieldElements.AnnotationText, Enumerations.Level.Reply, Enumerations.ValueType.Value));
+                    requestedMetadata.AddField(new IshRequestedMetadataField(FieldElements.AnnotationReplies, Enumerations.Level.Annotation, Enumerations.ValueType.Value));
+                    string xmlIshAnnotationReplies = IshSession.Annotation25.RetrieveMetadata(annotationIdsToDelete.ToArray(), "", requestedMetadata.ToXml());
+                    ishObjectsList.AddRange(new IshObjects(ISHType, xmlIshAnnotationReplies).Objects);
+                    var ishObjectsAnnotationReplies = ishObjectsList.FindAll(i => i.IshFields.GetFieldValue(FieldElements.AnnotationReplies, Enumerations.Level.Annotation, Enumerations.ValueType.Value) != "");
```

Comment:
I would adapt the filtering avoid that you need to filter on the client later. This can be achieved by 
1. adding a filter on reply level (e.g. FISHANNOTATIONCREATOR not empty)  because as soon as you add a filter on the reply level only annotations with replies will be returned
2. adding a filter on FieldElements.AnnotationReplies not empty instead of selecting the field and filter on the client later